src/hooks/useTokenBalances.ts

```diff
@@ -49,11 +50,11 @@ async function fetchBalancesForToken(
   }
 }
 
-async function _getBalances(walletInfo: WalletInfo): Promise<TokenBalanceDetails[] | null> {
+async function _getBalances(walletInfo: WalletInfo): Promise<TokenBalanceDetails[]> {
   const { userAddress, networkId } = walletInfo
   log('[useTokenBalances] getBalances for %s in network %s', userAddress, networkId)
   if (!userAddress || !networkId) {
-    return null
+    return []
```

Comment:
Are we sure we want to replace `null` with an empty string?

`null` meant, I don't know
`[]` I know but there aren't any tokens

Maybe in this case it's not too relevant, I would be OK leaving it like this, but just want to hear your thoughts

Reply:
It may be best, let me check the implementation again

Reply:
should it throw?

Reply:
one approach is to assert every important required param (`userAddress`, `contractAddress`) etc and handle life this way. Thoughts? @Velenir @alfetopito @anxolin

Reply:
the other approach, as was tackled in dx-react for example, was not blocking UI and rendering empty data sets (`[]` in this approach which renders the screen but with no usable params (`userADdress` etc) and then presenting an error somewhere to read. This could be approached with the `toast` things or a banner when a certain disconnect happens (like `dx-react`)

Reply:
Imo, not having a network or account is not an exception, it's a normal case, where the user doesn't have the wallet connected, this is why I think it's fine to signal that returning a `null` (would say, I don't know).

Note that the alternative of throwing wouldn't be good for a hook, since if you throw you are moving the responsibility to the caller the hook, and hooks cannot be conditionally rendered (one of their core restrictions)

In dx-react I think we weren't handing very well this states, this is why sometimes you could see balance 0 for some seconds, and then changing to something else (confusing for an user)